core/src/odbc_uri.rs

```diff
@@ -266,6 +273,7 @@ impl<'a> ODBCUri<'a> {
                 ServerAddress::parse(server).map_err(Error::InvalidClientOptions)?
             ])
             .credential(cred)
+            .app_name(self.get_app_name())
             .build())
     }
 }
```

Comment:
It would be good to have a unit test below to validate that the `APPNAME` behavior is as expected.

Reply:
Good catch! I had one but for some reason didn't include it in the commit. I've updated it to ensure the mongodb uri overrides the odbc uri, if both are provided.